### PR TITLE
Added /profileDiffs

### LINKED ISSUE #521: GET /profileDiffs endpoint
API to get the documents from collection profileDiffs whose approval is "PENDING"

Refer here : https://github.com/Real-Dev-Squad/identity-service/issues/18

### PR DESCRIPTION
**Pull request by lakshayman**
For return pending profile diffs

Closes #521

### CODE REVIEW

File: models/profileDiffs.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const firestore = require('../utils/firestore')
+const profileDiffsModel = firestore.collection('profileDiffs')
+
+/**
+ * Fetches the pending profile diffs
+ * @param query { Object }: Filter for profile diffs data
```
Comment (shubham-y) [inline models/profileDiffs.js:6]: It seems the parameter is not required. Can we please remove it?

Comment (lakshayman) [reply models/profileDiffs.js]: Done removed it

File: controllers/profileDiffs.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  try {
+    const pendingProfileDiffs = await profileDiffsQuery.fetchProfileDiffs();
+
+    return res.json({
+      message: "Profile Diffs returned successfully!",
+      profileDiffs: pendingProfileDiffs,
+    });
+  } catch (error) {
+    logger.error(`Error while fetching profile diffs: ${error}`);
+    return res.boom.serverUnavailable("Something went wrong please contact admin");
```
Comment (harshith-venkatesh) [inline controllers/profileDiffs.js:19]: Could we not look into having global constants for serverUnavailable and general error message?

File: models/profileDiffs.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const firestore = require("../utils/firestore");
+const profileDiffsModel = firestore.collection("profileDiffs");
+
+/**
+ * Fetches the pending profile diffs
+ * @return {Promise<profileDiffsModel|Array>}
+ */
+const fetchProfileDiffs = async () => {
+  try {
+    const snapshot = await profileDiffsModel.where("approval", "==", "PENDING").get();
```
Comment (harshith-venkatesh) [inline models/profileDiffs.js:10]: Could we look into creating constants here

File: models/profileDiffs.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    const profileDiffs = [];
+    snapshot.forEach((doc) => {
```
Comment (harshith-venkatesh) [inline models/profileDiffs.js:12]: Are we handling situation like undefined here?

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: test/integration/profileDiffs.test.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      .request(app)
+      .get("/profileDiffs")
+      .set("cookie", `${cookieName}=${superUserAuthToken}`)
+      .end((error, response) => {
+        if (error) {
+          return done(error);
+        }
+
+        expect(response).to.have.status(200);
+        expect(response.body.message).to.be.equal("Profile Diffs returned successfully!");
```
Comment (shubham-y) [inline test/integration/profileDiffs.test.js:45]: Can we add assertions to check each key of profileDiff object?

File: test/integration/profileDiffs.test.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+  before(async function () {
+    newUserId = await addUser(newUser);
+    newUserAuthToken = authService.generateAuthToken({ userId: newUserId });
+
+    superUserId = await addUser(superUser);
+    superUserAuthToken = authService.generateAuthToken({ userId: superUserId });
+  });
+
+  it("Should return pending profileDiffs, using authorized user (super_user)", function (done) {
```
Comment (shubham-y) [inline test/integration/profileDiffs.test.js:34]: Can we add a test for empty profileDiffs response?

File: models/profileDiffs.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+const firestore = require("../utils/firestore");
+const profileDiffsModel = firestore.collection("profileDiffs");
+
+/**
+ * Fetches the pending profile diffs
+ * @return {Promise<profileDiffsModel|Array>}
+ */
+const fetchProfileDiffs = async () => {
```
Comment (shubham-y) [inline models/profileDiffs.js:8]: **Suggestion:** Add a unit test for this model function

**Review by shubham-y — CHANGES_REQUESTED** [review]
Good work
Requesting some changes

File: models/profileDiffs.js (hunk 4 above)
Comment (lakshayman) [reply models/profileDiffs.js:12]: yes

File: test/integration/profileDiffs.test.js (hunk 5 above)
Comment (lakshayman) [reply test/integration/profileDiffs.test.js:45]: We have only message that we can check

File: models/profileDiffs.js (hunk 7 above)
Comment (lakshayman) [reply models/profileDiffs.js:8]: Can't write as pallab wrote a few functions in his PR after merging we will add that.
